proxy-node-shared/src/main/java/uk/gov/ida/notification/shared/metadata/MetadataCertsPublishingResource.java

```diff
@@ -67,51 +77,36 @@ public MetadataSigningCertsView getMetadataSigningCerts() {
                     uriInfo.getBaseUri(), metadataPublishPath), e);
         }
 
-        final File concatenatedPemCertsFile = new File(metadataCACertsFilePath.toString());
-        if (!concatenatedPemCertsFile.exists()) {
-            throw new InvalidMetadataException(format(
-                    "No file containing metadata signing certs found at {0}", metadataCACertsFilePath));
-        }
-
-        String concatenatedPemCerts;
-        try {
-            concatenatedPemCerts = Files.readString(concatenatedPemCertsFile.toPath());
-        } catch (IOException e) {
-            throw new InvalidMetadataException(format(
-                    "Couldn't read metadata certs file from {0}", metadataCACertsFilePath), e);
-        }
+        final String concatenatedPemCerts = readFile(metadataCACertsFilePath).orElseThrow(() ->
+                new InvalidMetadataException("Couldn't read metadata certs file from " + metadataCACertsFilePath));
 
         final List<Cert> certificates = Arrays
                 .stream(concatenatedPemCerts.split(BEGIN_LINE))
                 .filter(s -> !s.isBlank())
                 .map(c -> c.substring(0, c.indexOf(END_LINE)))
-                .map(c -> c.replace("\n", ""))
-                .map(String::trim)
-                .filter(c -> !c.equals(metadataSigningCertBase64.replace("\n", "").trim()))
+                .map(c -> BEGIN_LINE + c + END_LINE)
                 .map(this::createCert)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
                 .collect(Collectors.toList());
 
+        readFile(metadataSigningCertFilePath).flatMap(this::createCert).ifPresentOrElse(
```

Comment:
A comment here might be good to explain what the stream is doing. It looks like it's removing the metadata signing cert from the list of concatenated pem certs?

Reply:
Yep that's what it does 👍

Reply:
A comment would help for people reading it in future.

Reply:
Sure 👍